amgut/handlers/secondary_survey.py

```diff
@@ -33,6 +33,12 @@ def get(self):
         form = survey_class()
         if survey_id:
             form.process(data=sec_survey.fetch_survey(survey_id))
+
+        # load existing information into the form
+        if survey_id != '':
```

Comment:
Maybe, but since I don't have any unit-tests (still waiting for data in the database) in place, I don't want to change working code only for better style right now.